src/auto/credentials.rs

```diff
@@ -0,0 +1,107 @@
+// This file was generated by gir (5a68ad0) from gir-files (469db10)
+// DO NOT EDIT
+
+use Error;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::fmt;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct Credentials(Object<ffi::GCredentials, ffi::GCredentialsClass>);
+
+    match fn {
+        get_type => || ffi::g_credentials_get_type(),
+    }
+}
+
+impl Credentials {
+    pub fn new() -> Credentials {
+        unsafe {
+            from_glib_full(ffi::g_credentials_new())
+        }
+    }
+}
+
+impl Default for Credentials {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl fmt::Display for Credentials {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", CredentialsExt::to_string(self))
+    }
+}
+
+pub trait CredentialsExt {
+    //fn get_native(&self, native_type: /*Ignored*/CredentialsType) -> /*Unimplemented*/Option<Fundamental: Pointer>;
+
+    #[cfg(any(feature = "v2_36", feature = "dox"))]
+    fn get_unix_pid(&self) -> Result<i32, Error>;
+
+    fn get_unix_user(&self) -> Result<(), Error>;
```

Comment:
Strange getter.

Reply:
It `u32` and GIR must only convert to none `void` and `bool` types. Very strange

Reply:
And it really not on Windows: I get `undefined reference to 'g_credentials_get_unix_user'` when try to use.